library/alloc/src/vec/mod.rs

```diff
@@ -1787,9 +1822,24 @@ impl<T, A: Allocator> Vec<T, A> {
     /// }
     /// ```
     ///
+    /// Deallocating a vector using [`Box`] (which uses [`dealloc`] internally):
+    ///
+    /// ```
+    /// use std::mem::{ManuallyDrop, MaybeUninit};
+    ///
+    /// let mut v = ManuallyDrop::new(vec![0, 1, 2]);
```

Comment:
Maybe it'd be a better example to use types that are `Drop` and run `drop_in_place` with the right amount of data before deallocating. Otherwise one would cause leaks by following this pattern.